### PR TITLE
RDEV-5859 Fix loader utils vulnerability

### PR DESCRIPTION
**Pull request by Vivokas20**
Fix Loader-utils and json5 Snyk vulnerability:
[https://outsystemsrd.atlassian.net/browse/RDEV-5859](url)
[https://outsystemsrd.atlassian.net/browse/RDEV-5861](url)

Since the most recent loader utils still used the json5 with the vulnerability, as it was a transitive reference we can manually bump it in our project instead as they suggest on their repo https://github.com/OutSystems/monaco-editor/pull/14

### CODE REVIEW

**Review by OS-pedromiguelcarvalho — DISMISSED** [review]

**Comment by os-lmo** [comment]
@Vivokas20 , did you validate that, importing json5 in our package.json, the version being used is that instead of the one referenced in the load-utils, package? I believe I validated a similar scenario some time ago and it did not work.
By the way, the comment that you mentioned seems not clarify that...

File: webpack-plugin/package-lock.json [hunk 1]
```diff
@@ -1,8 +1,1788 @@
...
+				"webpack": "^5.64.1",
+				"webpack-cli": "^4.9.1"
+			},
+			"peerDependencies": {
+				"monaco-editor": ">= 0.31.0",
+				"webpack": "^4.5.0 || 5.x"
+			}
+		},
+		"node_modules/@discoveryjs/json-ext": {
+			"version": "0.5.5",
```
Comment (os-lmo) [inline webpack-plugin/package-lock.json:31]: Most of the changes on this file are weird. Can you double check (reverting and doing a npm install again) if they are correct?

File: webpack-plugin/package.json [hunk 2]
```diff
@@ -43,6 +43,7 @@
 		"webpack-cli": "^4.9.1"
 	},
 	"dependencies": {
-		"loader-utils": "^2.0.2"
+		"loader-utils": "^2.0.4",
+		"json5": "2.2.3"
```
Comment (os-lmo) [inline webpack-plugin/package.json:47]: As I pointed out, I am in doubt that this works. Did you validate this?

**Review by os-lmo — COMMENTED** [review]
See my comments

**Comment by os-lmo** [comment]
Just a last question: since these packages are for webpack (used in dev time), are they a security concern for us? I assume they are not used in the distribution...

File: webpack-plugin/package-lock.json (hunk 1 above)
Comment (Vivokas20) [reply webpack-plugin/package-lock.json:31]: Yes, the I tried to do npm install again and got the same result. The first time we did an npm install in the root package.json we also made a lot of changes as you can see here https://github.com/OutSystems/monaco-editor/pull/8/files which is why I didn't think it was strange

File: webpack-plugin/package.json (hunk 2 above)
Comment (Vivokas20) [reply webpack-plugin/package.json:47]: I checked the node_modules folder that was generated and the json5 version was the correct one. Then just to be sure here is the npm list for the json5 dependency where we can see that the version used by loader-utils is the correct one.
image

Comment (Vivokas20) [reply webpack-plugin/package.json:47]: This method will only work for updates up to the next major version, maybe this is why when you tried using this method it didn't work

**Comment by Vivokas20** [comment]
> Just a last question: since these packages are for webpack (used in dev time), are they a security concern for us? I assume they are not used in the distribution...

You are right and this was something we discussed on our team yesterday. In fact we don't use these packages. But since from what I understand in Snyk you can disregard the vulnerability but only up to 3 months. Since this change is a small one and it's not very costly we think it's best to just fix it instead of having to go through the process of disregarding it on Snyk every 3 months.

**Pull request closed without merging**